extension/content_script.js

```diff
@@ -635,21 +640,64 @@ function toggleExtensionVisibility() {
 
 function setupListenerForOnOff() {
 	chrome.runtime.onMessage.addListener(	
-		function(request, sender, sendResponse) {	
-			if (request.command === "toggleUI") {		
+		function(request, sender, sendResponse) {
+			function toggleApp() {
 				if (doc === null) {
 					// Make sure these load after animation
-					$("#loadingIcon").show(100, function() {
+					$("#loadingIcon").show(500, function() {
 						preprocessPage();
 						toggleExtensionVisibility();
 					});	
 				}
 				else { toggleExtensionVisibility(); }
+			}			
+			if (request.command === "toggleUI") {
+				toggleApp();
+			}	
+			// TODO: This can be perhaps refactored. For now, it's an easy fir
+			else if (request.command === "toggleUITutorial") {
+				if (settings.getAppStatus()) {
+					return; // If app status is on, do nothing
+				} 
+				toggleApp();
```

Comment:
The tutorial has different cases for if the persistent app status is on or off. This handles it, though perhaps not elegantly.